src/ifcgeom/mapping/IfcCurveSegment.cpp

```diff
@@ -376,11 +316,59 @@ class curve_segment_evaluator {
 		auto dx = dr[0] / m;
 		auto dy = dr[1] / m;
 
-		eval_ = [px, py, dx, dy](double u) {
-			auto x = px + u * dx;
-			auto y = py + u * dy;
-			return Eigen::Vector3d(x, y, 0);
-			};
+		if (segment_type_ == ST_HORIZONTAL) {
+
+			eval_ = [px, py, dx, dy](double u) {
+				auto x = px + u * dx;
+				auto y = py + u * dy;
+				Eigen::VectorXd vec(4);
+				vec << x, y, 0.0, 1.0;
+				return vec;
+				};
+
+		}
+		else if (segment_type_ == ST_VERTICAL) {
+
+			eval_ = [py, dy](double u) {
+				auto z = py + u * dy;
+				Eigen::VectorXd vec(4);
+				vec << 0.0, 0.0, z, 1.0;
+				return vec;
+				};
+
+		}
+	}
+
+	void operator()(IfcSchema::IfcPolynomialCurve* p) {
+
+		if (segment_type_ == ST_HORIZONTAL) {
+			auto coeffX = p->CoefficientsX();
```

Comment:
@aothms horizontal not finished yet since polynomial is not typically. For vertical, profile typically use a parabolic curve. I've been using IfcPolynomialCurve assumption Cy1*u^2 + Cy2*u + Cy3. However, the ifc specification is silent regarding how the coefficients should be applied. It could be Cy1 + Cy2*u + Cy3*u^2. I'm also assuming the polynomial curve is in a plane (2D curve). 

Any thoughts or suggestions?

Reply:
Maybe @peterrdf do you have time for this quick question?

Reply:
Based on [this thread](https://forums.buildingsmart.org/t/ifcpolynomialcurve-clarification/4716) in bSI forums it appears that it should be the latter interpretation of coefficient order.  Or at least that is what GeometryGym appears to be using.  I'd be happy to test the implementation against the data that exists for synthetic test cases from the IfcRail project.  I would just need a compiled version of the python extension as my C++ skills are generally in the range of `Hello, world!`.  Or perhaps I could implement github actions on my fork for compiling? @aothms any recommendations here?  I have VS 2022 Community for Win but much more comfortable in WSL2 Ubuntu.

Reply:
On Tuesday (10th of October) we have a meeting with somebody also getting different values for Clothoid. Let me check this first, there could be very well an issue in the examples also, and then come back here with a clean explanation for calculating the points of a clothoid based on some of the theoretical examples as we have available on GitHub (https://github.com/bSI-RailwayRoom/IFC-Rail-Unit-Test-Reference-Code/tree/master/alignment_testset).

Also let me then try to make a basic example based on IfcClothoid itself (i.e. without trimming).

Reply:
@civilx64 here are some instructions I put together for building with vs2022. https://github.com/RickBrice/PGSuperIfcExtensions/blob/BIM_for_Bridges_TPF/devdocs/Dependencies.md